cf/test/test_Data_utils.py

```diff
@@ -0,0 +1,108 @@
+import datetime
+import faulthandler
+import unittest
+
+import dask.array as da
+import numpy as np
+
+faulthandler.enable()  # to debug seg faults and timeouts
+
+import cf
+
+
+class DataUtilsTest(unittest.TestCase):
+    def test_Data_Utils__da_ma_allclose(self):
+        """TODO."""
+        # Create a range of inputs to test against.
+        # Note that 'a' and 'a2' should be treated as 'allclose' for this
+        # method, the same result as np.ma.allclose would give because all
+        # of the *unmasked* elements are 'allclose', whereas in our
+        # Data.equals method that builds on this method, we go even further
+        # and insist on the mask being identical as well as the data
+        # (separately, i.e. unmasked) all being 'allclose', so inside our
+        # cf.Data objects 'a' and 'a2' would instead *not* be considered equal.
+        a = np.ma.array([1.0, 2.0, 3.0], mask=[1, 0, 0])
+        a2 = np.ma.array([10.0, 2.0, 3.0], mask=[1, 0, 0])
+        b = np.ma.array([1.0, 2.0, 3.0], mask=[0, 1, 0])
+        c = np.ma.array([1.0, 2.0, 100.0], mask=[1, 0, 0])
+        d = np.array([1.0, 2.0, 3.0])
+        e = a + 5e-04  # outside of tolerance to set, namely rtol=1e-05
+        f = a + 5e-06  # within set tolerance to be specified, as above
+
+        # Test the function with these inputs as both numpy and dask arrays...
+        allclose = cf.data.dask_utils._da_ma_allclose
+        da_ = da.from_array(a)
+
+        self.assertTrue(allclose(a, a).compute())
```

Comment:
Do we need to test `_da_ma_allclose` for non-dask array inputs?

Reply:
I think the function is fine, just that the `a` in `self.assertTrue(allclose(a, a).compute())` is a numpy array, not a dask array. Apologies if I'm missing something

Reply:
Ah, I see - in that case I will amend the test. Thanks.